examples/Ballot.bl

```diff
@@ -0,0 +1,89 @@
+version 1.1
+
+contract Ballot {
+
+    struct Voter {
+        int weight
+        bool voted
+        address delegate
+        int vote
+    }
+
+    struct Proposal {
+        String name
+        int voteCount
+    }
+
+    address chairperson
+
+    Map<address, Voter> voters
+
+    Proposal[] proposals
+
+    constructor(String[] proposalNames) {
+        chairperson = msg.sender
+        voters[chairperson].weight = 1
+        proposals = new Proposal[proposalNames.length]
+
+        foreach(i, String name : proposalNames) {
+            proposals[i] = new Proposal(name, 0)
+        }
+    }
+
+    [Pre=msg.sender == chairperson]
+    [Pre=!voters[voter].voted]
+    [Pre=voters[voter].weight == 0]
+    function void giveRightToVote(address voter) {
+        voters[voter].weight = 1
+    }
+
+    [Pre = !voters[msg.sender].voted]
+    [Pre = to != msg.sender]
+    function void delegate(address to) {
+        Voter sender = voters[msg.sender]
+        to_ = to
```

Comment:
`to` and `to_` can lead to careless mistakes.
I would rename
* `to` --> `initialTo`
* `to_` --> `currentTo`

Reply:
no type required for to_ or currentTo? I'm fine with that, as we know to is an address

Reply:
We don't support type inference in the first version. Therefore type is also required here.